File: brasil_burger/src/main/java/sn/ism/brasilburger/entity/Complement.java
package sn.ism.brasilburger.entity;

import sn.ism.brasilburger.entity.enums.TypeComplement;
import java.time.LocalDateTime;

public class Complement {
    private int id;
    private String nom;
    private TypeComplement type;
    private double prix;
    private String image;
    private boolean archive;
    private LocalDateTime dateCreation;
    private LocalDateTime dateModification;


    public Complement() {}

    public Complement(String nom, TypeComplement type, double prix, String image) {
        this.nom = nom;
        this.type = type;
        this.prix = prix;
        this.image = image;
        this.archive = false;
    }

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public String getNom() { return nom; }
    public void setNom(String nom) { this.nom = nom; }

    public TypeComplement getType() { return type; }
    public void setType(TypeComplement type) { this.type = type; }

    public double getPrix() { return prix; }
    public void setPrix(double prix) { this.prix = prix; }

    public String getImage() { return image; }
    public void setImage(String image) { this.image = image; }

    public boolean isArchive() { return archive; }
    public void setArchive(boolean archive) { this.archive = archive; }

    public LocalDateTime getDateCreation() { return dateCreation; }
    public void setDateCreation(LocalDateTime dateCreation) { this.dateCreation = dateCreation; }

    public LocalDateTime getDateModification() { return dateModification; }
    public void setDateModification(LocalDateTime dateModification) { this.dateModification = dateModification; }
    @Override
    public String toString() {
        return String.format("Complement{id=%d, nom='%s', type='%s', prix=%.2f FCFA, archive=%s}",
            id, nom, type, prix, archive ? "Oui" : "Non");
    }
}

File: src/Entity/Complement.php
<?php
namespace App\Entity;

use App\Repository\ComplementRepository;
use App\Enum\TypeComplement;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: ComplementRepository::class)]
#[ORM\Table(name: 'complement')]
#[ORM\Index(name: 'idx_complement_type', columns: ['type'])]
#[ORM\Index(name: 'idx_complement_archive', columns: ['archive'])]
#[ORM\HasLifecycleCallbacks]
class Complement
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name: 'id_complement')]
    private ?int $idComplement = null;

    #[ORM\Column(length: 100)]
    private ?string $nom = null;

    #[ORM\Column(length: 20, enumType: TypeComplement::class)]
    private ?TypeComplement $type = null;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2)]
    private ?string $prix = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $image = null;

    #[ORM\Column(type: 'boolean')]
    private ?bool $archive = false;

    #[ORM\Column(name: 'date_creation', type: 'datetime')]
    private ?\DateTimeInterface $dateCreation = null;

    #[ORM\Column(name: 'date_modification', type: 'datetime')]
    private ?\DateTimeInterface $dateModification = null;

    #[ORM\OneToMany(mappedBy: 'complement', targetEntity: LigneCommande::class)]
    private Collection $lignesCommande;

    public function __construct()
    {
        $this->dateCreation = new \DateTime();
        $this->dateModification = new \DateTime();
        $this->archive = false;
        $this->lignesCommande = new ArrayCollection();
    }

    #[ORM\PreUpdate]
    public function preUpdate(): void
    {
        $this->dateModification = new \DateTime();
    }

    public function getIdComplement(): ?int
    {
        return $this->idComplement;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;
        return $this;
    }

    public function getType(): ?TypeComplement
    {
        return $this->type;
    }

    public function setType(TypeComplement $type): self
    {
        $this->type = $type;
        return $this;
    }

    public function getPrix(): ?float
    {
        return $this->prix ? (float) $this->prix : null;
    }

    public function setPrix(float $prix): self
    {
        $this->prix = (string) $prix;
        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(?string $image): self
    {
        $this->image = $image;
        return $this;
    }

    public function getArchive(): ?bool
    {
        return $this->archive;
    }

    public function setArchive(bool $archive): self
    {
        $this->archive = $archive;
        return $this;
    }

    public function getDateCreation(): ?\DateTimeInterface
    {
        return $this->dateCreation;
    }

    public function getDateModification(): ?\DateTimeInterface
    {
        return $this->dateModification;
    }

    public function getLignesCommande(): Collection
    {
        return $this->lignesCommande;
    }
}
